src/flamenco/runtime/program/fd_native_cpi.c

```diff
@@ -57,8 +57,7 @@ fd_native_cpi_execute_system_program_instruction( fd_exec_instr_ctx_t * ctx,
         if( acct_meta->is_writable ) {
           instr_info->acct_flags[j] |= FD_INSTR_ACCT_FLAGS_IS_WRITABLE;
         }
-        // TODO: should check the parent has signer flag set
-        if( acct_meta->is_signer ) {
+        if( acct_meta->is_signer && fd_instr_acc_is_signer_idx( ctx->instr, k ) ) {
```

Comment:
This passes a transaction account index into fd_instr_acc_is_signer_idx which takes in an instruction account index